src/rendergraph/common/attributeset.cpp

```diff
@@ -0,0 +1,12 @@
+#include "rendergraph/attributeset.h"
+
+using namespace rendergraph;
+
+AttributeSet::AttributeSet(std::initializer_list<Attribute> list,
+        const std::vector<QString>& names)
+        : backend::AttributeSet(list, names) {
+}
+
+const std::vector<Attribute>& AttributeSet::attributes() const {
+    return m_attributes;
+}
```

Comment:
"code duplicated between the backends should be in the common layer"

AttributeSet is a bit of a special case, and it might be that the common layer contains too much stuff that is only relevant for the opengl backend. But it was not trivial to come up with an implementation that serves as a frontend for both.

Reply:
> "code duplicated between the backends should be in the common layer"


makes sense. Then the stuff in the uniform(set) and attribute(set) code could basically be completely be in common afaict

Reply:
ah yes, I already cleaned that up.